src/main/lrsql/system/lrs.clj

```diff
@@ -375,4 +376,37 @@
     (let [conn  (lrs-conn this)
           input (react-input/delete-reaction-input reaction-id)]
       (jdbc/with-transaction [tx conn]
-        (react-cmd/delete-reaction! backend tx input)))))
+        (react-cmd/delete-reaction! backend tx input))))
+  rp/StatementReactor
+  (-react-to-statement [this statement-id]
+    (let [conn (lrs-conn this)
+          statement-results
+          (jdbc/with-transaction [tx conn]
+            (reduce
+             (fn [acc {:keys [reaction-id
+                              error]
+                       :as   result}]
+               (if error
+                 (let [input (react-input/error-reaction-input
+                              reaction-id error)]
+                   (react-cmd/error-reaction! backend tx input)
+                   acc)
+                 (conj acc (select-keys result [:statement :authority]))))
+             []
+             (:result
+              (react-q/query-statement-reactions
+               backend tx {:trigger-id statement-id}))))]
+      ;; Submit statements one at a time with varying authority
+      {:statement-ids
+       (reduce
+        (fn [acc {:keys [statement authority]}]
+          (into acc
+                (:statement-ids
+                 (lrsp/-store-statements
```

Comment:
i dont see where it would loop from reactions, but actually i guess that flow is what you're working on now? Possibly triggered from store-statements?

Reply:
Also I see what you mean about store-statements needing to repeat due to authority, but to be fair batch prep isnt ACTUALLY that efficient in lrsql. The db is still called n times.

Reply:
> Possibly triggered from store-statements?

Yep, the current plan is to do a non-blocking submit to the queue (channel) from store-statements after everything is done, conditional on this feature being enabled ofc.